engine/commands/cortex_upd_cmd.cc

```diff
@@ -0,0 +1,167 @@
+          return;
+        }
+        CTL_INF("Matched variant: " << matched_variant);
+
+        for (auto& asset : assets) {
+          auto asset_name = asset["name"].get<std::string>();
+          if (asset_name == matched_variant) {
+            std::string host{"https://github.com"};
+
+            auto full_url = asset["browser_download_url"].get<std::string>();
+            std::string path = full_url.substr(host.length());
+
+            auto fileName = asset["name"].get<std::string>();
+            CTL_INF("URL: " << full_url);
+
+            auto download_task = DownloadTask{.id = "cortex",
+                                              .type = DownloadType::Cortex,
+                                              .error = std::nullopt,
+                                              .items = {DownloadItem{
+                                                  .id = "cortex",
+                                                  .host = host,
+                                                  .fileName = fileName,
+                                                  .type = DownloadType::Cortex,
+                                                  .path = path,
+                                              }}};
+
+            DownloadService download_service;
+            download_service.AddDownloadTask(
+                download_task,
+                [this](const std::string& absolute_path, bool unused) {
+                  // try to unzip the downloaded file
+                  std::filesystem::path download_path{absolute_path};
+                  CTL_INF("Downloaded engine path: " << download_path.string());
+
+                  std::filesystem::path extract_path =
+                      download_path.parent_path().parent_path();
+
+                  archive_utils::ExtractArchive(download_path.string(),
+                                                extract_path.string());
+
+                  // remove the downloaded file
+                  // TODO(any) Could not delete file on Windows because it is currently hold by httplib(?)
+                  // Not sure about other platforms
+                  try {
+                    std::filesystem::remove(absolute_path);
+                  } catch (const std::exception& e) {
+                    CTL_WRN("Could not delete file: " << e.what());
+                  }
+                  CTL_INF("Finished!");
+                });
+            break;
+          }
+        }
+      } catch (const nlohmann::json::parse_error& e) {
+        std::cerr << "JSON parse error: " << e.what() << std::endl;
+        return;
+      }
+    } else {
+      CTL_ERR("HTTP error: " << res->status);
+      return;
+    }
+  } else {
+    auto err = res.error();
+    CTL_ERR("HTTP error: " << httplib::to_string(err));
+    return;
+  }
+#if defined(_WIN32)
+  std::string temp = ".\\cortex_tmp.exe";
+  remove(temp.c_str());  // ignore return code
+
+  std::string src =
```

Comment:
small nit, apply `std::filesystem::path` for concatenation path will simplify the code.